### ISSUE TITLE
22.0.0.2 & Bundler activation

### ISSUE THREAD

**Issue by MSP-Greg** [oracle/truffleruby#2586]
Puma is failing on TruffleRuby jobs.  This seemed to happen when TruffleRuby went from 21.3.0 to 22.0.0.2.  It may also involve recent changes to ruby/setup-ruby.  Not.Sure.

Passing job using 21.3.0 - https://github.com/puma/puma/runs/4938565379

Failing job using 22.0.0.1 - https://github.com/puma/puma/runs/4943479989

Both jobs use the same workflow, and both install Bundler 2.3.5, but the failing job logs
```
You have already activated bundler 2.2.22, but your Gemfile requires bundler 2.3.5.
```

**Comment by eregon** [comment]
Thank you for the report, @bjfish and I will take a look.
I suspect it's something new in Bundler or in RubyGems that broke this.

"your Gemfile requires bundler 2.3.5." sounds confusing, I guess it's just because the Gemfile.lock was created with Bundler 2.3.5.

Maybe not using the latest Bundler release (e.g., just use Bundler 2.2.22 shipped with TruffleRuby) would help here.
That's not possible with current setup-ruby (bundler: none means bundler-cache: true has no effect), but that should be easily fixable.

**Comment by eregon** [comment]
FWIW this worked: https://github.com/ruby/setup-ruby/runs/4942414592?check_suite_focus=true
truffleruby 22.0 + bundler 2.3.5 for bundle install + bundle exec. Weird.
I'll support `bundler: none + bundler-cache: true` in setup-ruby so at least we have a workaround.

**Comment by eregon** [comment]
Workaround in https://github.com/puma/puma/pull/2811 but we'll keep investigating on how to fix Bundelr 2.3.5+ and TruffleRuby 22.0/latest.

**Comment by eregon** [comment]
This is fixed on truffleruby-head with https://github.com/oracle/truffleruby/commit/752a2f3f496087b0424d5bf5ec166e36176aeaa4.

For 22.0 I did the workaround of always using the shipped Bundler for TruffleRuby in ruby/setup-ruby:
https://github.com/ruby/setup-ruby/releases/tag/v1.97.0

Ideally rake-compiler would fix this and then the workaround would probably not be needed anymore.
